pkg/daemon/httpchecker.go

```diff
@@ -25,27 +26,35 @@ import (
 )
 
 // WaitForServerReady waits until the HTTP(S) server can respond to a GET request. Should NOT allow cancelling the retry as shuting down non-ready server may cause deadlock.
-func WaitForServerReady(serverAddr string, insecureSkipVerify bool) error {
+func WaitForServerReady(serverAddr string, insecureSkipVerify bool, clientCertEnabled bool) error {
 
 	t := http.DefaultTransport.(*http.Transport).Clone()
 	t.TLSClientConfig = &tls.Config{}
 	client := &http.Client{Transport: t}
 
-	var url string
+	var targetUrl string
 	if insecureSkipVerify {
 		t.TLSClientConfig.InsecureSkipVerify = true
-		url = "https://" + serverAddr
+		targetUrl = "https://" + serverAddr
 	} else {
-		url = "http://" + serverAddr
+		targetUrl = "http://" + serverAddr
 	}
 
 	get := func() error {
-		resp, err := client.Get(url)
-		if err == nil {
+		resp, err := client.Get(targetUrl)
+		if err != nil {
+			// if client certificate disabled, return ALL errors.
+			// if client certificate enabled, return ALL errors but exclude client certificate verification error.
+			if !clientCertEnabled || errors.Unwrap(err).Error() != "remote error: tls: certificate required" {
```

Comment:
```suggestion
			errCause := errors.Unwrap(err)
			if !clientCertEnabled || (errCause !=nil && errCause.Error() != "remote error: tls: certificate required") {
```
Sorry that I missed out the `nil` case.
Please help to add the `nil` checking and try to simplify/refactor it.

Reply:
fixed: [db6688b](https://github.com/AthenZ/k8s-athenz-sia/pull/174/commits/db6688b5d775175e4fd54af7466cadc33ebca87d)